test/kdump_test.rb

```diff
@@ -159,4 +167,290 @@
+      Yast::Kdump.ReadKdumpKernelParam
+    end
+
+    let(:is_crashkernel_set) { Yast::Kdump.crashkernel_param }
+    let(:crashkernel_needed) { Yast::Kdump.add_crashkernel_param }
+    let(:several_ranges) { Yast::Kdump.crashkernel_list_ranges }
+    let(:allocated_mem) { Yast::Kdump.allocated_memory }
+
+    context "when the param is not present" do
+      let(:kernel_param) { :missing }
+
+      it "reports param as not found" do
+        expect(is_crashkernel_set).to eq false
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "does not schedule writing on the param" do
+        expect(crashkernel_needed).to eq false
+      end
+    end
+
+    context "when the param is set to true" do
+      let(:kernel_param) { :present }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+    end
+
+    context "when the param is a number" do
+      let(:kernel_param) { "32M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "correctly reads the size" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+
+    context "when the param is a range" do
+      let(:kernel_param) { "64M-:32M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "does not find several ranges" do
+        expect(several_ranges).to eq false
+      end
+
+      it "correctly reads the size" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+
+    context "when the param includes several ranges" do
+      let(:kernel_param) { "-200M:32M,200M-:64M" }
+
+      it "reports presence of the param" do
+        expect(is_crashkernel_set).to eq true
+      end
+
+      it "schedules the writing on the param" do
+        expect(crashkernel_needed).to eq true
+      end
+
+      it "finds several ranges" do
+        expect(several_ranges).to eq true
+      end
+
+      it "correctly reads the size of the last range" do
+        expect(allocated_mem).to eq "32"
+      end
+    end
+  end
+
+  describe ".WriteKdumpBootParameter" do
+    context "during autoinstallation" do
+      before do
+        allow(Yast::Mode).to receive(:autoinst).and_return true
+        allow(Yast::Mode).to receive(:update).and_return false
+        allow(Yast::Mode).to receive(:normal).and_return false
```

Comment:
better is to use `Yast::Mode.SetMode("autoinstallation")` as it is single line which is obvious what it does